### DISCUSSION TITLE
v3.2.2

### DISCUSSION THREAD

**Discussion by danielroe** [nuxt/nuxt#19123] [Announcements]
> **3.2.1** is a patch release with (lots of) bug fixes and performance improvements since last week's minor release.
> **3.2.2** was a swiftly following release to patch an issue with `nuxi init`

## 👀 Highlights

As a patch release, there are mostly bug fixes and performance improvements in the changelog. (Nevertheless, it's always worth reading through!) But one point of note is an experimental reload strategy when chunk errors are encountered. We're hoping to finalise the API and land it in v3.3 (our next feature release) with https://github.com/nuxt/nuxt/pull/19086, but you can test out an experimental version with the following config:

```ts
export default defineNuxtConfig({
  experimental: {
    emitRouteChunkError: 'reload'
  }
})
```

With this strategy, your app will hard reload on route changes if there's a chunk error. More info at https://github.com/nuxt/nuxt/pull/19038.

## 👉 Changelog

[compare changes](https://github.com/nuxt/nuxt/compare/v3.2.0...v3.2.1)

### 🚀 Enhancements

  - **nuxt:** Add experimental `app:chunkError` hook and reload strategy ([#19038](https://github.com/nuxt/nuxt/pull/19038))

### 🔥 Performance

  - **nuxt:** Allow tree-shaking empty meta from build ([#19032](https://github.com/nuxt/nuxt/pull/19032))
  - **nuxt:** Animate transform rather than width ([#19073](https://github.com/nuxt/nuxt/pull/19073))
  - **nuxt:** Don't include side-effects from `#components` ([#19008](https://github.com/nuxt/nuxt/pull/19008))

### 🩹 Fixes

  - **schema:** Allow type inference of arrays in runtime config ([#18931](https://github.com/nuxt/nuxt/pull/18931))
  - **nuxt:** Avoid recursive import in nitro renderer ([#18948](https://github.com/nuxt/nuxt/pull/18948))
  - **nuxt:** Expose `nuxt/schema` subpath for augmentation ([#18922](https://github.com/nuxt/nuxt/pull/18922))
  - **nitro:** Ensure ssr error `statusCode` is a number ([#19001](https://github.com/nuxt/nuxt/pull/19001))
  - **nuxt:** Prevent hyphens forming child routes & warn if dupes are detected ([#18944](https://github.com/nuxt/nuxt/pull/18944))
  - **schema:** Transpile `nuxt/app` by default ([#19009](https://github.com/nuxt/nuxt/pull/19009))
  - **vite:** Exclude `nuxt/app` from optimised deps ([9e789c76c](https://github.com/nuxt/nuxt/commit/9e789c76c))
  - **vite:** Respect `isCustomElement` config for jsx transform ([#19053](https://github.com/nuxt/nuxt/pull/19053))
  - **nuxt:** Use parser to treeshake client-only declarations ([#18951](https://github.com/nuxt/nuxt/pull/18951))
  - **cli:** Pass through exit code from test errors ([#18959](https://github.com/nuxt/nuxt/pull/18959))
  - **nuxt:** Show client error if no page matches after validate fails ([#18978](https://github.com/nuxt/nuxt/pull/18978))
  - **cli:** Read `devServer` options from nuxt config ([#19055](https://github.com/nuxt/nuxt/pull/19055))
  - **nuxt:** Validate no `//` in path when constructing payload url ([#19085](https://github.com/nuxt/nuxt/pull/19085))
  - **nuxt:** Restore previous check on payload url ([e9ff34ace](https://github.com/nuxt/nuxt/commit/e9ff34ace))
  - **nuxt:** Test generated pathname ([af55b9882](https://github.com/nuxt/nuxt/commit/af55b9882))
  - **nuxt:** Respect redirects which differ only by trailing slash ([#18593](https://github.com/nuxt/nuxt/pull/18593))
  - **nuxt:** Pass nuxt + workspace paths when importing builder ([#19099](https://github.com/nuxt/nuxt/pull/19099))

### 💅 Refactors

  - **nuxt:** Combine imports from same relative path ([ee2f568fc](https://github.com/nuxt/nuxt/commit/ee2f568fc))

### 📖 Documentation

  - Add info for `nuxi devtools` command ([#18888](https://github.com/nuxt/nuxt/pull/18888))
  - Update link to bridge repo ([#18950](https://github.com/nuxt/nuxt/pull/18950))
  - Fix typos ([#18976](https://github.com/nuxt/nuxt/pull/18976))
  - Fix typo ([#18975](https://github.com/nuxt/nuxt/pull/18975))
  - Fix link to ESM page ([#19000](https://github.com/nuxt/nuxt/pull/19000))
  - Fix spacing ([#19018](https://github.com/nuxt/nuxt/pull/19018))
  - Flatten top-level project in layers example ([#18967](https://github.com/nuxt/nuxt/pull/18967))
  - Remove reference to obsolete `static` property ([80f73d39c](https://github.com/nuxt/nuxt/commit/80f73d39c))
  - Add note about nested directory order ([#19061](https://github.com/nuxt/nuxt/pull/19061))
  - Updated public directory explanation ([#18464](https://github.com/nuxt/nuxt/pull/18464))
  - Add information on when mw runs & how to skip ([#19083](https://github.com/nuxt/nuxt/pull/19083))
  - Add `sendRedirect` usage ([#19070](https://github.com/nuxt/nuxt/pull/19070))

### 📦 Build

  - **nuxt:** Add `nuxt/schema` to externals ([e3e768873](https://github.com/nuxt/nuxt/commit/e3e768873))
  - Add `nuxt` to nuxt externals ([b1b908b1e](https://github.com/nuxt/nuxt/commit/b1b908b1e))

### 🏡 Chore

  - Increase node 14 minor version constraint ([#19111](https://github.com/nuxt/nuxt/pull/19111))
  - Update nitropack and unjs dependencies ([#19100](https://github.com/nuxt/nuxt/pull/19100))
  - Dedupe rollup ([576ce9ee3](https://github.com/nuxt/nuxt/commit/576ce9ee3))

### ✅ Tests

  - Skip vnode warning on windows dev mode ([71bcd9550](https://github.com/nuxt/nuxt/commit/71bcd9550))

### 🤖 CI

  - Run webpack/vite and dev/prod as matrices ([#18905](https://github.com/nuxt/nuxt/pull/18905))

### ❤️  Contributors

- Daniel Roe <daniel@roe.dev>
- Pooya Parsa <pyapar@gmail.com>
- Mathieu Magalhaes <mathieu.magalhaes@gmail.com>
- Yuyin 
- Hamish Claxton <hamishclaxton@gmail.com>
- Nathan Chase <nathan@nathanchase.com>
- Alex <self@pirsig.net>
- Julien Huang <julien.huang@outlook.fr>
- Nick Williams 
- Favour Felix Chinemerem 
- Alois Sečkár <ellrohir@seznam.cz>
- Florent Delerue <florentdelerue@hotmail.com>
- Xin Du (Clark) <clark.duxin@gmail.com>
- Tim Van Den Eijnden 
- Benjamin Flesch <bf@deutsche-cyberberatung.de>
- Lioness100 <jchickmm2@gmail.com>
- Felix

<hr /><em>This discussion was created from the release <a href='https://github.com/nuxt/nuxt/releases/tag/v3.2.2'>v3.2.2</a>.</em>

**Comment by ChuanfengLai** [comment]
Error: The requested module '/pikcloud/node_modules/.vite/deps/vue.js?v=1d1d91db' does not provide an export named 'Component'
```
- Operating System: `Windows_NT`
- Node Version:     `v16.19.1`
- Nuxt Version:     `3.2.2`
- Nitro Version:    `2.2.2`
- Package Manager:  `npm@8.19.3`
- Builder:          `vite`
- User Config:      `nitro`, `routeRules`, `typescript`, `runtimeConfig`, `app`, `css`, `build`, `modules`, `alias`, `vite`
- Runtime Modules:  `@pinia/nuxt@0.4.3`
- Build Modules:    `-`
```

**Reply by danielroe** [reply]
Please raise an issue with a reproduction 🙏

**Reply by ChuanfengLai** [reply]
> Please raise an issue with a reproduction 🙏

Fixed in v3.2.3.
